### PR TITLE
[WIP] Override list using table.

### PR DESCRIPTION
**Pull request by Anu1601CS**
Override list using table. `#__template_overrides`

See #39

### CODE REVIEW

File: administrator/components/com_templates/View/Template/HtmlView.php [hunk 1]
```diff
@@ -145,6 +145,7 @@ public function display($tpl = null)
 		$this->template    = $this->get('Template');
 		$this->preview     = $this->get('Preview');
 		$this->pluginState = PluginHelper::isEnabled('installer', 'override');
```
Comment (laoneo) [inline administrator/components/com_templates/View/Template/HtmlView.php]: Do we still need this variable?

**Comment by astridx** [comment]
We have to decide what we want to show in this column. 

1. If we show the last modify of the file, then we have to update it each time we edit the file. In this case I would not store it in database, I would read it form file system
2. If we want to show the date of last update, then we should rename the column title.

templates customise cassiopeia test administration 10

**Comment by astridx** [comment]
I am not sure if two ids in a document are really ok. We have the adminform id now twice in view /administrator/index.php?option=com_templates&view=template 

`<form action="/gsoc18/administrator/index.php?option=com_templates&amp;task=template.copy&amp;id=510…;file=L2h0bWwvY29tX2Fnb3Ntcy9hZ29zbS9kZWZhdWx0LnBocA" method="post" name="adminForm" id="adminForm"` 

and
 
`<form action="/gsoc18/administrator/index.php?option=com_templates&amp;view=template&amp;id=510&amp;file=L2h0bWwvY29tX2Fnb3Ntcy9hZ29zbS9kZWZhdWx0LnBocA" method="post" name="adminForm" id="adminForm">`

File: administrator/components/com_templates/View/Template/HtmlView.php (hunk 1 above)
Comment (Anu1601CS) [reply administrator/components/com_templates/View/Template/HtmlView.php]: Yes, if plugin is disabled. So, nothing will update in database right ? Then, we don't show tab.

File: administrator/components/com_admin/sql/others/mysql/utf8mb4-conversion-02.sql [hunk 2]
```diff
@@ -200,6 +201,7 @@ ALTER TABLE `#__redirect_links` DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_un
 ALTER TABLE `#__schemas` DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;
 ALTER TABLE `#__session` DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;
 ALTER TABLE `#__tags` DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;
+ALTER TABLE `#__template_overrides` DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;
```
Comment (astridx) [inline administrator/components/com_admin/sql/others/mysql/utf8mb4-conversion-02.sql]: Do we really need that? We created this table with this charset?

**Comment by Anu1601CS** [comment]
@astridx I am thinking like this.
`created_date` ----> When the update history is created.
`modified_date`-----> When this history again change or updated after the update then we update `modified_date`
This case occurs if user not check his updated overrides properly .
What are you thinking @astridx @laoneo  ?

File: administrator/components/com_admin/sql/others/mysql/utf8mb4-conversion-02.sql (hunk 2 above)
Comment (Anu1601CS) [reply administrator/components/com_admin/sql/others/mysql/utf8mb4-conversion-02.sql]: I am not sure about this. But, i saw that all table is listed here. So, i also added my table.

**Comment by astridx** [comment]
@Anu1601CS Yes. That is fine. But I would rename the title of the column. Because it is confusing if you edit the file and see an old date after viewing our view.

File: administrator/components/com_admin/sql/others/mysql/utf8mb4-conversion-02.sql (hunk 2 above)
Comment (laoneo) [reply administrator/components/com_admin/sql/others/mysql/utf8mb4-conversion-02.sql]: @wilsonge do we have to add new tables also here?

File: administrator/components/com_templates/View/Template/HtmlView.php (hunk 1 above)
Comment (laoneo) [reply administrator/components/com_templates/View/Template/HtmlView.php]: Ok, thanks.

File: administrator/components/com_templates/View/Template/HtmlView.php [hunk 3]
```diff
@@ -281,6 +282,11 @@ protected function addToolbar()
 			}
 		}
 
+		if (count($this->updatedList) !==0 && $this->pluginState)
```
Comment (laoneo) [inline administrator/components/com_templates/View/Template/HtmlView.php]: A space is missing after !==

**Comment by astridx** [comment]
Close as merged in https://github.com/joomla-projects/gsoc18_override_management/pull/39

**Pull request closed without merging**